url27a7ae3508_serkan_ozal_ocean_of_memories_tgz-pJ8-com_zeroturnaround_rebellabs_oceanofmemories_article1_objectlayout_ObjectMemoryLayoutDemoJ8/src/com/zeroturnaround/rebellabs/oceanofmemories/common/util/ClassUtil.java

```diff
@@ -0,0 +1,133 @@
+package com.zeroturnaround.rebellabs.oceanofmemories.common.util;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+
+import org.apache.log4j.Logger;
+
+public class ClassUtil {
+
+	private static final Logger logger = LogUtil.getLogger();
+	
+	public static final String CLASS_FILE_EXTENSION = ".class";
+	public static int BYTE_COUNT_IN_A_ROW_FOR_PRINT = 16; // Not final, it can be set directly
+	
+	private ClassUtil() {
+		
+	}
+	
+	public static String getPathOfClass(Class<?> clazz) {
+		return clazz.getResource(clazz.getSimpleName() + CLASS_FILE_EXTENSION).getPath();
+	}
+	
+	public static InputStream getInputStreamOfClass(Class<?> clazz) {
+		return clazz.getResourceAsStream(clazz.getSimpleName() + CLASS_FILE_EXTENSION);
+	}
+	
+	public static File getFileOfClass(Class<?> clazz) {
+		try {
+			return new File(clazz.getResource(clazz.getSimpleName() + CLASS_FILE_EXTENSION).toURI());
+		}
+		catch (URISyntaxException e) {
+			logger.error("Error occured while getting file of class " + clazz.getName(), e);
+			return null;
+		}
+	}
+	
+	public static byte[] getContentOfClass(Class<?> clazz) {
+		File classFile = getFileOfClass(clazz);
+		if (classFile.exists()) {
+			BufferedInputStream bis = null;
+			try {
+				bis = new BufferedInputStream(new FileInputStream(classFile));
+				byte[] buffer = new byte[(int)classFile.length()];
+				bis.read(buffer);
+				return buffer;
+			}
+			catch (IOException e) {
+				logger.error("Error occured while reading content of class " + clazz.getName(), e);
+			}
+			finally {
+				if (bis != null) {
+					try {
+						bis.close();
+					} 
+					catch (IOException e) {
+						logger.error(
+								"Error occured while closing file stream of class " + 
+								clazz.getName() + e);
+					}
+				}
+			}
+		}
+		else {
+			logger.error("Class file for path " + classFile.getAbsolutePath() + " couldn't be found");
+		}
+		return null;
+	}
+	
+	public static void printContentOfClassFile(Class<?> clazz) {
+		byte[] classContent = getContentOfClass(clazz);
+		if (classContent != null) {
+			for (int i = 0; i < classContent.length; i++) {
+	    		if (i % BYTE_COUNT_IN_A_ROW_FOR_PRINT == 0) {
+					System.out.print(String.format("[0x%04x]: ", i));
+				}
+	    		System.out.print(String.format("%02x ", classContent[i]));
+				if ((i + 1) % BYTE_COUNT_IN_A_ROW_FOR_PRINT == 0) {
+					System.out.println();
+				}
+	    	}	
+			System.out.println();
+		}
+	}
+	
+	public static void printContentOfClassFileByShowingPrintableCharacters(Class<?> clazz) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.